File: 프로그래머스/1/12982. 예산/예산.js
function solution(d, budget) {
    let answer = 0;
    let check = d.sort((a,b) => a-b).reduce((acc, cur) => {
        if((acc + cur <= budget)) {
            console.log(acc);
            console.log(cur);
            answer += 1;
            return acc + cur;
        }
    }, 0);
    
    return answer;
}

File: 프로그래머스/1/12982. 예산/예산.java
import java.util.*;

class Solution {
    public int solution(int[] d, int budget) {
        int answer = 0;
        int[] sortedArr = d;
        
        // 1. 오름차순 정렬
        Arrays.sort(sortedArr);
        
        // 2. 이전값 + 현재값 더한 값 <= 예산 이면, answer+1
        int sum = 0;
        for(int num : sortedArr) {
            if(sum + num <= budget) {
                sum = sum + num;
                answer += 1;
            }
        }
        
        return answer;
    }
}